interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const multipassSupportSummary = `allows operating as the Multipass service`
+
+const multipassSupportBaseDeclarationPlugs = `
+  multipass-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const multipassSupportBaseDeclarationSlots = `
+  multipass-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const multipassSupportConnectedPlugAppArmor = `
+# Description: allow operating as the Multipass daemon. This policy intentionally
+# allows the Multipass daemon to configure AppArmor, and we expect Multipass to
+# create and apply correct policies for the child processes it spawns.
+
+###include <abstractions/openssl>
+
+# Allow socket
+/run/multipass_socket  rw,
+capability chown,
+capability fsetid,
+
+# Multipass generates AppArmor profiles for the child processes it spawns
+# Allow it to add these profiles and run the child under that profile.
+/usr/sbin/aa-exec ixr,
+/sbin/apparmor_parser ixr,
+/etc/apparmor.d/cache/ r,
+/etc/apparmor.d/cache/.features r,
+/etc/apparmor.d/{,cache/}multipass* rw,
+/etc/apparmor.d/tunables/{,**} r,
+/etc/apparmor.d/abstractions/{,**} r,
+/etc/apparmor/parser.conf r,
+/etc/apparmor/subdomain.conf r,
+/sys/kernel/security/apparmor/.replace rw,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/{,**} r,
+
+capability net_admin,
+capability mac_admin,
+capability kill, # multipass needs to be able to kill child processes
+capability sys_module, # checking for kvm will involve loading module
```

Comment:
Hmm, I'm mildly sceptical about this. Is there any other way?

Reply:
Thanks for letting me know about the kmod interface in snapd, it should do the job. I'll update this when ready

Reply:
We discussed this briefly and I recommended to use the `kmod` backend where snapd does the module load for the snap.

Reply:
Replaced with kmod support in kvm